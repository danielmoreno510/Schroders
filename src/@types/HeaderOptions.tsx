/* eslint-disable */
type HeaderOptionsProps = {
  badge?: number;
  changeScreen(): any;
  isMenu: boolean;
};
