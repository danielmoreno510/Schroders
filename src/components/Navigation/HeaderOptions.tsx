import React from 'react';
import {TouchableOpacity} from 'react-native';
import {Icon} from 'native-base';

import styles from './styles';

const HeaderOptions: React.FC<HeaderOptionsProps> = ({
  changeScreen,
  isMenu,
}) => (
  <TouchableOpacity onPress={changeScreen} style={styles.headerLeftContainer}>
    <Icon
      type="MaterialIcons"
      name={isMenu ? 'close' : 'menu'}
      style={styles.iconColor}
    />
  </TouchableOpacity>
);

export default HeaderOptions;
