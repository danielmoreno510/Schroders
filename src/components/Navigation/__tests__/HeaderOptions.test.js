import React from 'react';
import TestRenderer from 'react-test-renderer';

import HeaderOptions from '../HeaderOptions';

describe('HeaderOptions component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      changeScreen: jest.fn(),
      position: 'left',
    };

    describe('WHEN the component is rendered', () => {
      const testRenderer = TestRenderer.create(
        <HeaderOptions {...testProps} />,
      );

      it('THEN should display a regular HeaderOptions', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
      });
    });

    describe('WHEN the component is rendered with badge', () => {
      const customProps = {...testProps, badge: 1, position: 'right'};
      const testRenderer = TestRenderer.create(
        <HeaderOptions {...customProps} />,
      );

      it('THEN should display a regular HeaderOptions', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
      });
    });
  });
});
