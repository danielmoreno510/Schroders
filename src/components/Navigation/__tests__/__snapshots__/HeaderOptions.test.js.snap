// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderOptions component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular HeaderOptions 1`] = `
"<TouchableOpacity onPress={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], contexts: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], withImplementation: [Function: bound withImplementation], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} style={{...}}>
  <Component type="MaterialIcons" name="menu" style={{...}} />
</TouchableOpacity>"
`;

exports[`HeaderOptions component GIVEN a set of valid props WHEN the component is rendered with badge THEN should display a regular HeaderOptions 1`] = `
"<TouchableOpacity onPress={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], contexts: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], withImplementation: [Function: bound withImplementation], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} style={{...}}>
  <Component type="MaterialIcons" name="menu" style={{...}} />
</TouchableOpacity>"
`;
